### PR TITLE
Refactor and simplify Storage.put

### PR DESCRIPTION
**Pull request by ace-e4s**
## Description
This PR moves all "put" logic to `storage.Storage.put`. In addition, `storage.Storage.put` is made to work with urls ++ only! All API calls moved to `client` methods.

We should abandon the use of REST api abstractions in the long run. So, new api calls are implemented directly in `client`. 

## Checklist
- [ ] PR title is descriptive and fit for injection into release notes (see tips below)
- [ ] Correct label(s) are used

PR title tips:
* Use imperative mood
* Describe the motivation for change, issue that has been solved or what has been improved - not how
* Examples:
  * Add functionality for Allan variance to sensor_4s.simulate
  * Upgrade to support Python 9.10
  * Remove MacOS from CI

### CODE REVIEW

File: datareservoirio/storage/storage.py [hunk 1]
```diff
@@ -39,36 +39,42 @@ def __init__(self, timeseries_api, files_api, downloader, uploader):
             Instance of files API.
         downloader: cls
             A strategy instance for handling downloads.
-        uploader: cls
-            A strategy instance for handling uploads.
+        auth : cls
+            An authenticated session that is used in all API calls. Must supply a
+            valid bearer token to all API calls.
+
```
Comment (vegard-solum-4ss) [inline datareservoirio/storage/storage.py:45]:
```suggestion
        session : cls
            An authenticated session that is used in all API calls. Must supply a
            valid bearer token to all API calls.


```

File: tests/test_storage/test_storage.py [hunk 2]
```diff
@@ -9,11 +9,9 @@
 import pytest
 
 from datareservoirio.appdirs import user_cache_dir
-from datareservoirio.storage import (
+from datareservoirio.storage import (  # BaseUploader,; DirectUpload,
```
Comment (vegard-solum-4ss) [inline tests/test_storage/test_storage.py:12]:
```suggestion
from datareservoirio.storage import

```

Comment (vegard-solum-4ss) [inline tests/test_storage/test_storage.py:12]:
```suggestion
from datareservoirio.storage import (

```

File: datareservoirio/storage/storage.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ class Storage:
     Handle download and upload of timeseries data in DataReservoir.io.
     """
 
-    def __init__(self, timeseries_api, files_api, downloader, uploader):
+    def __init__(self, timeseries_api, files_api, downloader, session):
         """
```
Comment (vegard-solum-4ss) [inline datareservoirio/storage/storage.py:30]: Can we remove files_api from init? Looks like it is not used anymore.

**Review by anna-follestad-4ss — DISMISSED** [review]
Please note possible bug posted in devDS chat, other than that - looks good!

**Review by vegard-solum-4ss — DISMISSED** [review]

**Comment by vegard-solum-4ss** [comment]
Have a look at my comments, other than that it looks good.

**Pull request merged**